### ISSUE TITLE
Deepvariant DataLoss Error

### ISSUE THREAD

**Issue by RewatiAtGit** [google/deepvariant#175]
Hello deepvariant community,
I am running deepvariant analysis on whole genome sequencing data. The first part of the analysis, the "make_examples" runs without error. However, in the "call_variants" step, I get the following error: 

![Screen Shot 2019-04-16 at 17 10 54](https://user-images.githubusercontent.com/9975286/56221509-aacef300-606a-11e9-9302-e3c7c2a9eed2.png)

My code is as follows:

N_SHARDS="8"  \
seq 0 $((N_SHARDS-1)) | parallel -k --line-buffer \
python ../bin/make_examples.zip \
--mode calling \
--ref /Volumes/workspace/input/reference/GRCh38/Homo_sapiens_hg38.fa  \
--reads /Volumes/workspace/input_march_mac7/AS-196110-LR-30685_HISATmapping/Aligned.out.bam \
--examples /Volumes/workspace/output_march_mac7/AS-196110-LR-30685.tfrecord@${N_SHARDS}.gz \
--gvcf /Volumes/workspace/output_march_mac7/AS-196110-LR-30685.gvcf.tfrecord@${N_SHARDS}.gz \
--task {}

python ../bin/call_variants.zip \
      --outfile /Volumes/workspace/output_march_mac7/AS-196110-LR-30685.vcf.gz \
      --examples /Volumes/workspace/output_march_mac7/AS-196110-LR-30685.tfrecord@8.gz \
      --checkpoint /Volumes/workspace/input/models/WGS/model.ckpt \
      --batch_size 32

However, when I run the same scripts on a BAM file generated using reads extracted from chromosome 1, all the steps run successfully.

Please let me know where I could be going wrong with the commands.

**Comment by pichuan** [comment]
Hi,
This can happen if you terminate things (ctrl-c) in the middle of a run, but sometimes the python processes are still running in the background. I also need to use top to inspect the jobs, and forcefully kill all DeepVariant jobs before I restart.
This is the only scenario that I've seen this happen before. if you might be encountering other cases where things can override each other (for example, accidentally running two things that write to the same place), this could also happen.
I'm going to close this issue, but if you think there might be something else going on, please let me know.

**Comment by ptrebert** [comment]
@pichuan I just stumbled upon the same thing, and it took me quite a while to figure out what's going on there. I am running deepvariant v0.9.0 (docker container), and I found that there is quite a lot of files left behind for failed jobs under /tmp on the execution host. Can you comment on deepvariant's behavior when two (or more) DV jobs are scheduled to the same execution host in a cluster setup? Should that be avoided?

**Comment by pichuan** [comment]
@ptrebert 
One clarifying question - Do you mean that you have multiple jobs that write to the same files at the same time, or multiple jobs that writes to different file names?

Previously, the issues was that if the user has some jobs they thought they killed (because they ctrl-c) but were still running in the background, this could cause the new jobs to write in the same files, and resulted in corrupted files.

If they're with different names, I think it should be fine. If you're seeing errors with different names, I can give it a try and report back. Let me know.

**Comment by ptrebert** [comment]
@pichuan Thanks for getting back at me about this. I am processing a single dataset, but each chromosome in a separate DeepVariant job (i.e., a single execution host in the cluster may run the jobs for, say, chr17, chr12 and chr5 in parallel). My question now is if it is guaranteed that the files created underneath /tmp have distinct names. To me, it just looked like consecutive numbers, which could be the cause for the DataLossError I observed. The output files for the different jobs are of course different (carrying the respective chromosome name), but what about the temp file?

**Comment by pichuan** [comment]
I just checked the code, and you're right that the temp file names will be the same:
https://github.com/google/deepvariant/blob/r0.9/scripts/run_deepvariant.py#L264-L266

For now, please pass in different `intermediate_results_dir` for each run.

For example:
`--intermediate_results_dir="/tmp/deepvariant_tmp_output/chr1"` for chr1, and so on.

I'll think about how we want to improve this in the future. I can think of a few options for future improvements, such as :

1. Use a random name for the internal /tmp files. Given that these are not exposed to the users anyway.
2. Use a unique name derived from the output VCF file, instead of calling all temp files the same name.

For now, using the `--intermediate_results_dir` should hopefully resolve your issue. Let me know if it works. If you have a suggestion on what's the best future improvement for better user experience, please let me know.

**Comment by ptrebert** [comment]
@pichuan 
Thanks for checking this. I am going to open a new issue about this linking to this one...
